src/Microsoft.ML.DataView/DataViewRowCursor.md

```diff
@@ -0,0 +1,218 @@
+﻿# `DataViewRowCursor` Notes
+
+This document includes some more in depth notes on some expert topics for
+`DataViewRow` and `DataViewRowCursor` derived classes.
+
+## `Batch`
+
+The `Batch` property is the mechanims by which the data from the multiple
+cursors returned by `IDataView.GetRowCursorSet` can be reconciled into a
+single, cohesive, sequence.
+
+`IDataView.GetRowCursorSet` can return a set of parallel cursors that
+partition the sequence of rows as would have normally been returned through a
+more typical call to `GetRowCursor`. These cursors can be accessed across
+multiple threads to enable parallel evaluation of a data pipeline. This is key
+for the data pipeline performance during featurization.
+
+However, even though the data pipeline can perform this parallel evaluation,
+at the end of this parallelization we sometimes want to recombine the separate
+thread's streams back into a single cohesive sequence. This can be done for
+several reasons: we may want repeatability and determinism, or the cursor may
+be stateful in some way that precludes the sharding inherent . But, for this end consumer that calls
+the single cursor. And, since a core `IDataView` design principle is
+repeatability, we now have a problem of how to reconcile those separate
+partitioninging. This reconciliation to re-sequence the partitioned dataset is
```

Comment:
Flows funny, plus a mis-spell. Recommend popping into word and doing a spell-check on this entire thing.

Reply:
Yeah, thanks Rogan. This paragraph got especially mangled, I've tried to finesse it a bit more. It's a bit of a difficult point...